### DISCUSSION TITLE
Help MeshCentral+Letsencrypt

### DISCUSSION THREAD

**Discussion by tecnicosya** [Ylianst/MeshCentral#5144] [General]
Hi guys, I install Meshcentral and work ok, but the letsencrypt dont work.
I try to get the certificate with the automatic process included on Meshcentral but not work.
I have Plesk and I installed Meshcentral in the same Server.
I want to use the certificates generated by Plesk on MeshCentral, but how?
The Script fail to generate the certificate bacause the plesk have nginx is using the 80 port

**Comment by dinger1986** [comment]
Are you using a proxy?

**Reply by tecnicosya** [reply]
> Are you using a proxy?

No. Use direct with proxy. But Plesk include Nginx and Apache proxy I don't use it for Meshcentral.

**Reply by dinger1986** [reply]
Ok. @si458 have the solution anyway

**Comment by si458** [comment]
you have to point meshcentral to where plesk stores there certificates `/usr/local/psa/var/certificates/` it would seem
https://talk.plesk.com/threads/ssl-certificate-files-location.336076/

**Reply by tecnicosya** [reply]
I found the certificates with extrange names I try to copy to meshcentral-data, rename the old certificates and rename the cryptic name to the name of old certificates.
And not work.
Or not is the right place or is not the right format.
From plesk admin I can get the certificate but in   .PEM format.
I generate a wildcard certificate from the main domain and for a subdomain. I have 2 certificates to test. And not work just copying and renaming.

**Reply by si458** [reply]
im not 100% sure im afraid, whats your `config.json`?
i think you need to copy the certifciate and the private key over and replace `webserver-cert-public.crt` and `webserver-cert-private.key`

**Reply by tecnicosya** [reply]
I found this, the certificate and the key are the same file. I need to split. 
        ssl_certificate             /opt/psa/var/certificates/scfglZCfy;
        ssl_certificate_key         /opt/psa/var/certificates/scfglZCfy;

**Comment by tecnicosya** [comment]
Finally solved!.
On Plesk Panel click on Letsencrypt certificates, advance options. 
Plesk show the Key and Certificate in text format.
I open the webserver-cert-private.key and replace the text with from LetsEncrypt and I  make the same with webserver-cert-public.crt.
Now my Meshcentral web panel  is working ok.
The only problem is not automatic the update but is ok for now.
